File: app/src/main/java/com/iceteaviet/fastfoodfinder/ui/storelist/StoreListActivity.java
package com.iceteaviet.fastfoodfinder.ui.storelist;

import android.os.Bundle;

import com.iceteaviet.fastfoodfinder.R;
import com.iceteaviet.fastfoodfinder.utils.DataUtils;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by MyPC on 11/30/2016.
 */
public class StoreListActivity extends AppCompatActivity {

    @BindView(R.id.rv_top_list)
    RecyclerView recyclerView;

    private StoreListAdapter adapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_store_list);

        ButterKnife.bind(this);

        adapter = new StoreListAdapter();
        LinearLayoutManager layoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);
        RecyclerView.ItemDecoration decoration = new DividerItemDecoration(getApplicationContext(), DividerItemDecoration.VERTICAL);
        recyclerView.addItemDecoration(decoration);
        loadData();
    }


    private void loadData() {
        adapter.setStores(DataUtils.getFakeStoreList());
    }
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/ui/storelist/StoreListActivity.kt
package com.iceteaviet.fastfoodfinder.ui.storelist

import android.os.Bundle
import androidx.annotation.Nullable
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.iceteaviet.fastfoodfinder.R
import com.iceteaviet.fastfoodfinder.ui.base.BaseActivity
import com.iceteaviet.fastfoodfinder.utils.getFakeStoreList
import kotlinx.android.synthetic.main.activity_store_list.*

/**
 * Created by MyPC on 11/30/2016.
 */
class StoreListActivity : BaseActivity() {
    lateinit var recyclerView: RecyclerView

    private var adapter: StoreListAdapter? = null

    override fun onCreate(@Nullable savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        recyclerView = rv_top_list

        adapter = StoreListAdapter()
        val layoutManager = LinearLayoutManager(applicationContext)
        recyclerView.layoutManager = layoutManager
        recyclerView.adapter = adapter
        val decoration = DividerItemDecoration(applicationContext, DividerItemDecoration.VERTICAL)
        recyclerView.addItemDecoration(decoration)
        loadData()
    }

    override val layoutId: Int
        get() = R.layout.activity_store_list


    private fun loadData() {
        adapter!!.setStores(getFakeStoreList())
    }
}
